controllers/api/booking-routes.js

```diff
@@ -1,8 +1,91 @@
-// http://url/api/booking/*
 const router = require("express").Router();
+const { Booking, User, Service } = require("../../models");
+const withAuth = require("../../utils/auth");
 
+// GET all bookings 
 
+router.get ('/', withAuth, async (req,res) => {
+    try {
+        const bookingData = await Booking.findAll({
+            include: [{ model: User}, { model: Service }],
+        });
 
+        res.status(200).json(bookingData);
+    } catch (err) {
+        res.status(500).json({message: 'An internal server error occurred 😿'});
+    }
+});
 
+// GET a single booking
+
+router.get ('/:id', withAuth, async (req,res) => {
+    try {
+        const bookingData = await Booking.findByPk(req.params.id, {
+            include: [{ model: User}, { model:Service }],
+        });
+
+    if(!bookingData) {
+        res.status(404).json({ message: 'No Booking with that id exists 😔'});
+        return;
+    }
+        
+    res.status(200).json(bookingData);
+    } catch (err) {
+        res.status(500).json({message: 'An internal server error occurred 😿'});
+    }
+    });
+
+// CREATE a booking 
+
+router.post('/create', withAuth, async (req,res) => {
+    try {
+        
+        const existingBooking = Booking.findAll({where: { service_id: req.body.service_id }});
+
+        if (existingBooking) {
+            res.status(400).json({
+                message: "A booking with this ID already exists",
+            });
+            return;
+        }
+        else {
+        const bookingCard = await Booking.create({
+
+            id: req.body.id
+            client_id: req.body.client_id,

```

Comment:
```suggestion
            client_id: req.session.user_id,

```

We want to try to use the user ID on the session as much as possible to prevent people from making bookings for different users by calling the API directly